luarules/gadgets/cus_gl4.lua

```diff
@@ -1541,12 +1541,12 @@ local function ProcessUnits(units, drawFlags, reason)
 			--under construction
 			--using processedUnits here actually good, as it will dynamically handle unitfinished and cloak on-off
 		else
-			if not spGetUnitIsCloaked(unitID) then
-				uniformcache[1] = 0
-				gl.SetUnitBufferUniforms(unitID, uniformcache, 12) -- cloak
-			end
 			--Spring.Echo("ProcessUnit", unitID, drawFlag)
 			if overriddenUnits[unitID] == nil then --object was not seen
+				if (not spGetUnitIsCloaked(unitID)) and Spring.ValidUnitID(unitID) then
```

Comment:
Shouldn't the validity check be first? else you can get an error from GetUnitIsCloaked that you're asking about an invalid unit
```suggestion
				if Spring.ValidUnitID(unitID) and not spGetUnitIsCloaked(unitID) then
```